src/server/api/user.ts

```diff
@@ -112,33 +131,19 @@ function validateState(
 }
 
 /**
- * Make sure all parameters needed for the pre-signed url request are present.
+ * Endpoint for a user to create a short URL. A short URL can either point to
+ * a long URL, or a file provided by the user.
  *
- * @param {string} fileType - File type of the file that is being uploaded.
- * This must be declared here so that subsequent PUT requests that use the
- * `Content-Type` header will pass header checks.
- * @param {string} key - Name of the entry to be inserted to the S3 bucket. Ensure
- * that the key does not collide with other files before declaring it here. Otherwise,
- * it will cause an existing file of the same key to be overridden.
+ * In order to create a file URL, provide the `isFile` parameter set to `true`, and supply
+ * the binary in a multipart/form-data under the field titled `file`. The longUrl
+ * parameter in the request body must point to the s3 bucket with the correct key. For
+ * example, a file upload request for the shortUrl named `test` should have `longUrl` set
+ * to `https://file[-staging].go.gov.sg/test`.
  */
-function validatePresignedUrlRequest(
-  req: Express.Request,
-  res: Express.Response,
-  next: Express.NextFunction,
-) {
-  if (!req.body.fileType || !req.body.key) {
-    return res.badRequest(
-      jsonMessage('Some or all required arguments are missing: fileType, key.'),
-    )
-  }
-  return next()
-}
-
-/**
- * Endpoint for a user to create a short URL.
- */
-router.post('/url', validateUrls, async (req, res) => {
+router.post('/url', fileUploadMiddleware, validateUrls, async (req, res) => {
   const { isFile, userId, longUrl, shortUrl } = req.body
+  // @ts-ignore Type definition does not know about the compulsory file field.
+  const file: UploadedFile | undefined = req.files?.file
```

Comment:
I see that the type of `req.files?.file` is  `fileUpload.UploadedFile | fileUpload.UploadedFile[] | undefined`. under what circumstances will it be an array? should we be addressing the potential types using type guards instead of overriding the compiler?

Reply:
Okay. Even though I haven't been able to reproduce the scenario where the field manifests as an array, adding type guards are ultimately the safer way to go.